File: platform/android/lynx_android/src/main/java/com/lynx/tasm/gesture/handler/NativeGestureHandler.java
// Copyright 2023 The Lynx Authors
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

package com.lynx.tasm.gesture.handler;

import androidx.annotation.NonNull;
import com.lynx.tasm.behavior.LynxContext;
import com.lynx.tasm.gesture.GestureArenaMember;
import com.lynx.tasm.gesture.detector.GestureDetector;

/**
 * The NativeGestureHandler class is a concrete implementation of the BaseGestureHandler class
 * for handling native gestures. It handles native gestures on android, iOS and other platforms
 */
public class NativeGestureHandler extends PanGestureHandler {
  /**
   * Constructs a NativeGestureHandler object with the specified properties.
   *
   * @param sign               the sign indicating the direction of the gesture
   * @param lynxContext        the LynxContext associated with the gesture handler
   * @param gestureDetector    the GestureDetector associated with the gesture handler
   * @param gestureArenaMember the GestureArenaMember associated with the gesture handler
   */
  public NativeGestureHandler(int sign, LynxContext lynxContext,
      @NonNull GestureDetector gestureDetector, GestureArenaMember gestureArenaMember) {
    super(sign, lynxContext, gestureDetector, gestureArenaMember);
  }
}

File: clay/ui/gesture_handler/handler/native_gesture_handler.cc
// Copyright 2026 The Lynx Authors. All rights reserved.
// Licensed under the Apache License Version 2.0 that can be found in the
// LICENSE file in the root directory of this source tree.

#include "clay/ui/gesture_handler/handler/native_gesture_handler.h"

#include "clay/ui/gesture_handler/handler/pan_gesture_handler.h"

namespace clay {

NativeGestureHandler::NativeGestureHandler(
    int sign, PageView *page_view,
    std::shared_ptr<GestureDetector> gesture_detector,
    fml::WeakPtr<GestureArenaMember> gesture_arena_member)
    : PanGestureHandler(sign, page_view, gesture_detector,
                        gesture_arena_member) {}

}  // namespace clay
